piqueserver/config.py

```diff
@@ -0,0 +1,86 @@
+import toml
+from pprint import pprint
+
+class ConfigException(Exception):
+    pass
+
+
+class ConfigEngine():
+    '''
+    configuration engine that handles loading and saving config,
+    and all the things
+    '''
+    def __init__(self):
+        self.raw_config = {}
+        self.style = 'TOML'
+
+    def load_config(self, config_file, style='TOML'):
+        # TODO: could also have this merge into current config if available
+        self.style = style
+
+        if style == 'TOML':
+            self.raw_config = toml.load(open(config_file))
+        else:
+            raise Exception()
+
+    def dump_config(self, out_file=None, style='TOML'):
+        if out_file is None:
+            pprint(self.raw_config)
+        # TODO
+        pass
+
+    def get(self, name, default):
```

Comment:
I'm kind of torn on this. On one hand a property would look a lot nicer in code, on the other hand it hides the fact that it's actually a function call.

Reply:
I was also thinking of something like the following to force read-only:

```
@property
def value():
  return self.value
```

(also since the value is able to be cached, the function call probably doesn't cost much)